### PR TITLE
#155736056 Apply CircleCI Integration (With Static Code Analysis Tools)

### PR DESCRIPTION
**Pull request by AnthonyGW**
#### What does this PR do?
Add a CircleCI config file (config.yml) that describes the pipeline for testing (CI) and deploying build files to a Slack channel.
Add the rulesets for the static code analysis tools (Findbugs, Checkstyle, PMD, Android Lint).

#### Description of Task to be completed?
The art-android project uses TravisCI for continuous integration (checking code quality, running both unit tests and integration tests).
Set up the project to use CircleCI for the same functions as well as a Continuous Deployment section in the pipeline. Consider the current git workflow.

#### How should this be manually tested?
Commits and pull requests pushed to github will trigger a new build. When the build succeeds or fails, a notification will be sent to the configured slack channel.

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/155736056
https://www.pivotaltracker.com/story/show/155703248

#### Screenshots
Slack message format:
screen shot 2018-03-09 at 12 10 47
screen shot 2018-03-09 at 12 11 15

### CODE REVIEW

**Comment by AnthonyGW** [comment]
The external environment variables (CIRCLE_TOKEN, SLACK_CHANNEL_HOOK & DEPLOYMENT_CHANNEL) cannot be set until the project is active on CircleCI.
Someone with admin privileges has to add the project.
An admin also has to approve the request for a Slack APP webhook.

File: app/src/debug/google-services.json [hunk 1]
```diff
@@ -15,6 +15,7 @@
       },
       "oauth_client": [
         {
```
Comment (AnthonyGW) [inline app/src/debug/google-services.json]: This one was just an empty line I missed while resolving merge conflicts in the web editor

**Comment by AnthonyGW** [comment]
The last 3 commits are for resolving merge conflicts with the updated develop (competing line changes)

File: app/src/debug/google-services.json (hunk 1 above)
Comment (arnoldokoth) [reply app/src/debug/google-services.json]: Cool @AnthonyGW

**Review by arnoldokoth — APPROVED** [review]

**Pull request merged**